src/handler.jl

```diff
@@ -72,13 +98,24 @@ function checkpoint(handler::Handler, name::String, data::Dict{Symbol}; tags...)
     end
 end
 
-#=
-Define our no-op conditions just to be safe
-=#
-function checkpoint(handler::Nothing, name::String, data::Dict{Symbol}; tags...)
+"""
+    DictHandler(objects)
+
+Saves checkpointed objects into a dictionary where the keys are strings generated from
+the checkpoint tags and name.
+"""
+struct DictHandler <: AbstractHandler
+    objects::Dict{String, Dict}
+    DictHandler() = new(Dict{String, Dict}())
+end
+
+commit!(handler::DictHandler, k::AbstractString, data) = setindex!(handler.objects, data, k)
+
+function checkpoint(handler::DictHandler, name::String, data::Dict{Symbol}; tags...)
+    # TODO: Remove duplicate wrapper code
     checkpoint_deprecation(tags...)
     with_checkpoint_tags(tags...) do
-        debug(LOGGER, "Checkpoint $name triggered, but no handler has been set.")
-        nothing
+        debug(LOGGER, "Checkpoint $name triggered, with context: $(join(CONTEXT_TAGS[], ", ")).")
+        handler.objects[getkey(handler, name)] = data
```

Comment:
Is thread safety a concern here? Should we include a lock in the handler?

Reply:
I don't think it's likely to be an issue given our current use-case. Though we may actually want to disallow overwriting the keys in general?

Reply:
Hmm, after reviewing some discord threads it looks like dicts in julia aren't threadsafe due to the overhead. FWIW, I don't think adding a lock is worth the complexity or performance hit here.